tags/TagMap.md

```diff
@@ -23,23 +23,78 @@ and group stats, annotate traces and logs.
+entities and they do not create additional hops. Every propagation implementation shoud support option 
+`decrementTTL` (default set to true) that allows proxies to set it to false.
+
+**For now, ONLY special values (0 and -1) are supported.**
+
+#### Special Values
+- **NO_PROPAGATION (0)**: Tag with `TagTTL` value of zero is considered to have local scope and
+ is used within the process it created.
+ 
+- **UNLIMITED_PROPAGATION (-1)**: A Tag with `TagTTL` value of -1 can propagate unlimited hops.
+ However, it is still subject to outgoing and incoming (on remote side) filter criteria. 
+ See `TagPropagationFilter` in [Tag Propagation](#Tag Propagation). `TagTTL` value of -1
+ is typical used to represent a request, processing of which may span multiple entities.
+
+#### Example for TagTTL > 0
+On a server side typically there is no information about the caller besides ip/port,
+but in every process there is a notion of "service_name" tag that is added as a "caller" tag before
+serialization when a RPC/HTTP call is made. For the "caller" tag, desirable `TagTTL` value is 1.
+
+Note that TagTTL value of 1 is not supported at this time. The example is listed here simply to
+show a possible use case for TagTTL > 0.
+ 
+### Processing at Receiver and Sender
+For the sake of completeness, processing of `Tag` and `TagTTL` at sender and receiver includes
+the values other than the special values of `TagTTL`.
+
+#### At Receiver
+Upon receiving a tag from remote entity a tag extractor
+
+- MUST decrement the value of `TagTTL` by one if it is greater than zero.
+- MUST not change the value of `TagTTL` if it is -1.
+- MUST treat the value of `TagTTL` as -1 if is not present.
+- MUST discard the `Tag` for any other value of `TagTTL`.
+
+#### At Sender
+Upon preparing to send a tag to a remote entity a tag injector
+- MUST send the tag AND include `TagTTL` if its value is greater than 0.
+- MUST send the tag without 'TagTTL' if its value is -1. Absence of TagTTL on the wire is treated as having TagTTL of -1.
+  This is to optimize on-the-wire representation of common case.
+- MUST not send the tag if the value of `TagTTL` is 0.
+
+A tag accepted for sending/receiving based on `TagTTL` value could still be excluded from sending/receiving based on
+`TagPropagationFilter`.
+
+## Tag Conflict Resolution
+If a new tag conflicts with an existing tag then the new tag takes precedence. Entire `Tag` along 
+with `TagValue` and `TagMetadata` is replaced by the most recent tag (regardless of it is locally
+generated or received from a remote peer). Replacement is limited to a scope in which the 
+conflict arises. When the scope is closed the orignal value prior to the conflict is restored.
+For example,
+```
+Enter Scope 1
+   Current Tags T1=V1, T2=V2
+    Enter Scope 2
+      Add Tags T3=V3, T2=v4
+      Current Tags T1=V1, T2=V4, T3=V3 <== Value of T2 is replaced by V4.
+    Close Scope 2
+   Current Tags T1=V1, T2=V2  <== T2 is restored.
```

Comment:
Maybe point that metadata is also restored.

Reply:
sure.